docs/HOL_Continuous-Integration/README.md

```diff
@@ -219,116 +42,116 @@ the page.
+![](<media/build_working_directory_integration.png>)
 
-**8.** Select the second Gradle task and **edit the task name** to say
-*OrderService* and set the **Gradle Wrapper** to the following location:
-(NOTE: set the Options to **-x text** as the test have external dependencies on a mongo database.)
+**8.** Select the second Gradle task and **click the pencil icon** to edit the task name. Name the task *OrderService* and set the **Gradle Wrapper** to the following location:
 
 	src/Backend/OrderService/gradlew
-	-x test
+
+![](<media/build_gradle_order.png>)
 	
+**9.** Since Order Service does have unit tests in the project, we can automate running the tests as part of the build by adding in `test` in the **Gradle tasks** field. Keep the **JUnit Test Results** checkbox to "Publish to VSO/TFS" checked, and set the **Test Results Files** field to `**/TEST-*.xml`. 
 
-Set the **Working Directory** to the following location:
+Expand the **Advanced** section, and set the **Working Directory** to the following location:
 
 	src/Backend/OrderService
 
-![](<media/build_gradle_order.png>)
+![](<media/build_working_directory_order.png>)
 
-**9.** Select the third Gradle task and **edit the task name** to say *Clients*
-and set the **Gradle Wrapper** to the following location:
+**10.** Select the third Gradle task and **click the pencil icon** to edit the task name. Name the task *Clients* and set the **Gradle Wrapper** to the following location:
 
 	src/Clients/gradlew
 
-Set the **Working Directory** to the following location:
+![](<media/build_gradle_clients.png>)
 
-	src/Clients
+**11.** Uncheck the checkbox in **JUnit Test Results** to Publish to VSO/TFS since we will not be running automated tests in Clients. Expand the **Advanced** section, and set the **Working Directory** to the following location:
 
-![](<media/build_gradle_clients.png>)
+	src/Clients
 
-**10.** Click **Add build step...** and add a **Publish Build Artifacts** task
+![](<media/build_working_directory_clients.png>)
 
+**9.** Click **Add build step...**, select the **Utility** tab, and add two **Copy and Publish Build Artifacts** tasks.
 
 ![](<media/build_add_pub_step.png>)
 
-**11.** Select the Publish Build Artifacts task, and fill in the input values
-with the following:
+**10.** Select the first Publish Build Artifacts task, and fill in the input values with the following:
 
-	Copy Root: $(build.sourcedirectory)
-	Contents: **/build/libs/!(buildSrc)*.?ar
+	Copy Root: $(Build.SourcesDirectory)
+    Contents: **/build/libs/!(buildSrc)*.?ar
 	Artifact Name: drop
 	Artifact Type: Server
 
 ![](<media/build_pub_step_details.png>)
 
-**NOTE:** The Content field supports minimatch patterns. You can find more information here: [https://github.com/Microsoft/vso-agent-tasks](https://github.com/Microsoft/vso-agent-tasks "https://github.com/Microsoft/vso-agent-tasks")
+**11.** Select the second Publish Build Artifacts task, and fill in the input values with the following:
```

Comment:
Do we need this part in this HOL ?
My guess is to talk about that in the CD HOL